libethereum/Client.cpp

```diff
@@ -1163,11 +1163,18 @@ h256 Client::importTransaction( Transaction const& _t ) {
         gasBidPrice = this->gasBidPrice();
     }
 
+    bool multitransactionMode = checkMultitransactionMode();
     Executive::verifyTransaction( _t,
         bc().number() ? this->blockInfo( bc().currentHash() ) : bc().genesis(), state,
-        *bc().sealEngine(), 0, gasBidPrice );
+        *bc().sealEngine(), 0, gasBidPrice, multitransactionMode );
+
+    ImportResult res;
+    if ( multitransactionMode && state.getNonce( _t.sender() ) < _t.nonce() ) {
+        res = m_tq.import( _t, IfDropped::Ignore, true );
+    } else {
+        res = m_tq.import( _t );
+    }
```

Comment:
why would we need two separate ways to import "current" and "future" transaction? can't the transaction queue determine by itself where to put new txn? (and in fact it first puts every new txn into "current" queue and then conditionally moves it into the "future" queue!)